app/main.py

```diff
@@ -1,3 +1,27 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    memory_cache = {}
+
+    def inner(*args, **kwargs) -> list and int:
+        nonlocal memory_cache
+        immute = (dict, set, list)
+        for argumets in args:
+            if type(argumets) in immute:
+                return
+
+        memory_cache_key = ""
+        input_num = args[0] if type(args[0]) is tuple else args
+        for each in input_num:
+            memory_cache_key += str(each) + "$"
+        memory_cache_key += str(args[1])
```

Comment:
No need to create some memory cache key, u can use args as a key for ur memory_cache dict, cause the args always immutable